### ISSUE TITLE
Strings listed even though they're not supposed to be translated

### ISSUE THREAD

**Issue by missmatsuko** [enupal/translate#19]
### Description
May be related to #18 and #5, I'm finding class names listed in Translations.

I definitely don't want this happening, because I want Translations to be available to content authors, and they could break the site layout by translating class names.

I'm having a hard time figuring out which template(s) these strings are coming from because:
1. Most of my templates are in a folder, and the templates listing only shows the folder and not the template files within it.
2. I'm using helper classes, so these classes are used widely throughout the templates within the folder.

So I guess there's multiple issues here:
1. Strings that are not filtered with `|t` are appearing in translations.
2. Templates in folders aren't listed in the Templates listing, only the first child within the templates directory is listed. e.g. `templates/components/` instead of `templates/components/foo.twig` and `templates/components/bar.twig`
3. I cannot hide/lock strings that should not be translated between languages so that content authors can safely edit only a subset of approved static translation strings.

### Steps to reproduce
I can't provide more details at this time. Let me know if there's a way to debug the problem further.

### Additional info

- Craft version: 3.0.34
- PHP version:
- Database driver & version:
- Plugin version: 1.1.8

**Comment by Bergrebell** [comment]
same here.

in one of our twig-files there is this string:

`{% set layoutClass = 'pageBuilder-imageSingle-is' ~ block.layout|capitalize %}`

and the plugin extracts 

` object-fit: { objectFit };` as a translation.

**Comment by andrelopez** [comment]
Hi folks,

It looks like the regular expression used by Enupal Translate need to be updated. I'll look into this over the weekend and I hope to have the new regex working by the end of the next week.

Best,

**Comment by missmatsuko** [comment]
Also noticed some of my more complex translation strings are not getting captured. Don't know if it's because of the complexity or just the new lines?

e.g. 
```
{{ '{
    num,
    plural,
    =0{ # results }
    =1{ # result }
    other{ # results }
    } for “{query}”'|t({
    num: locations|length,
    query: mapQuery,
}) }}
```

**Comment by missmatsuko** [comment]
Played with the regex a little. It looks like the current regex will match any filter containing `t`.

e.g. `{{ 'whatever'|trim }}` `{{ 'whatever'|split }}` `{{ 'whatever'|whatever }}` all match.

Adding a `/s` flag to the end of the Regex enables capturing across multiple lines, so the example in my previous comment would get captured.

I'm not sure what all the valid translation filters/functions are. Are there tests set up for that?

For Twig, maybe this works?
https://regexr.com/474q1

**Comment by andrelopez** [comment]
@missmatsuko  I love your regex, the only problem with that regex is that does not accept more than one filter to the right:

`{{ 'example'|uppercase|t }}` 

however if the `|t` it's always the first filter it works:

`{{ 'example'|t|uppercase }}` 

Please let me know if you can find a solution to this issue and I'll consider adding your regex 👍

**Comment by missmatsuko** [comment]
@andrelopez I don't think it would be useful to capture strings with filters applied before the translate filter (e.g. `{{ 'example'|uppercase|t }}`) with a RegEx search anyways. The captured string would still be 'example', but the translation string would need the filter applied to become 'EXAMPLE' for the translation to actually process as expected, and I don't think this plugin handles that.

It would be more accurate if the plugin were hooking into the actual functions somehow, rather than a RegEx search across files.

Overall, I think my RegEx is an improvement over the current one since it's not capturing all filters containing a `t` anymore.

**Comment by andrelopez** [comment]
Hi @missmatsuko @Bergrebell 

Thanks for all the feedback. We just released Enupal Translate v1.2.0 where we added a regex search API. So as twig|html has complex scenarios, we have added a `Twig search method` setting under `Enupal Translate -> Settings -> General` where you can choose the search method. Currently, there are two methods available 
 - **Legacy search method:** this is the old method where some twig string were captured as false positives.
 - **Optimized search method:** this is the new regex method suggested by @missmatsuko where in my tests looks much better than the Legacy method and fixes several false positive strings reported by the users.

In the future, I'm considering allowing users to register their own regex to search translatable strings in the different file extensions: php, js, twig. @missmatsuko  here is the optimized class example that we added to the plugin: https://github.com/enupal/translate/blob/master/src/integrations/OptimizedTwigSearch.php

FYI:
- This update requires Craft 3.1
- Users that upgrade the plugin will default the Legacy twig search method (can be updated under the general settings)
- New installations will have the Optimized twig method as default.
- If you're using the Optimized method please make sure your translate twig filter be the first in your strings, e.g `{{ 'Hello world'|t|upper }}`

Please let me know if you have any questions.

Thank you!
